### PR TITLE
4679 javascript bundle

### PR DESCRIPTION
**Pull request by JayanthyChengan**
## Related Issues

- connects to 
#4679

### CODE REVIEW

**Comment by JayanthyChengan** [comment]
@mheppler  noticed under My Data tab, there are still English terms not toggling with language selection
https://localhost:8080/dataverseuser.xhtml?selectTab=dataRelatedToMe

I am wondering where does these come from?
screen shot 2018-12-05 at 3 13 27 pm

**Comment by mheppler** [comment]
@JayanthyChengan I've got some answers for you on where to locate those text examples you've highlighted above.

Account pg > My Data
- "1 to 10 of 67 Results" comes from the template file found at `/webapp/mydata_templates/result_message_only.html`
- Publication status labels like "Published" and "Draft" can be found at `/webapp/mydata_templates/cards_minimum.html`
- Roles labels like "Admin" and "Curator" is coming from the `getSelectedFilterParamsAsJSON()` function which can be found in the `MyDataFinder.java` which pulls it from the database, which you can get from the bundle with the db name as the key like you've done previously in other beans
- "DRAFT VERSION" in the citation is coming the Solr index which treats it the same as "data" which doesn't get translated

Let me know if you have any more My Data or javascript questions. It's been a long while since I've looked at this code, but I'd be glad to help sort through it all again to get you pointed in the right direction.

**Comment by JayanthyChengan** [comment]
@mheppler 
Item 1 and Item 2:
noticed that mydata_templates is using Nunjucks templating ( it's new to me ) , do I need to follow  
https://github.com/KSDaemon/nunjucks-intl to pass bundle variables?

Item 3:
To get role names from bundle, tried the below snippet, but the language is not toggling in MyDataFinder.java when selected from language dropdown list.

`public JsonArrayBuilder getListofSelectedRoles(){
        
        JsonArrayBuilder jsonArray = Json.createArrayBuilder();
        
        for (Long roleId : this.filterParams.getRoleIds()){
            try {
                System.out.println("=========role." + this.rolePermissionHelper.getRoleName(roleId).toLowerCase().replace(" ", "") + ".name");
                String role = BundleUtil.getStringFromPropertyFile("role." + this.rolePermissionHelper.getRoleName(roleId).toLowerCase().replace(" ", "") + ".name", "BuiltInRoles");
                jsonArray.add(role);
            } catch (Exception e)
            {
                jsonArray.add(this.rolePermissionHelper.getRoleName(roleId));
            }

        }
        return jsonArray;                
    }`

**Comment by mheppler** [comment]
@JayanthyChengan 

Yes, after reviewing the options with the dev team, the best way to get bundle text into the My Data `html` template files is by passing variables. I would add them to `mydata_fragment.xhtml`, where the `init_mydata_page()` function is called.

As for the role names, I have asked @scolapasta to provide some feedback for you.

**Comment by scolapasta** [comment]
@JayanthyChengan 
The first thing I'd try is adding the jsf:view tag to the user page like we had added for the alert messages. (basically it seems to me that the page is not getting the right locale)

**Comment by JayanthyChengan** [comment]
@mheppler - I tried by passing hidden variables from mydata_fragment.xhtml , and used it in mydata.js to toggle using bundle value.  Please check in this commit
https://github.com/IQSS/dataverse/pull/5351/commits/507c82f35c76a252ff7c401b5c67c34b4bffc657?utf8=%E2%9C%93&diff=split&w=1

I noticed that from this link =>  http://localhost:8080/dataverseuser.xhtml?selectTab=dataRelatedToMe , all tabs are working fine.  

But  if I click on "Edit Account -> Account Information or Password" on the right side ,   I see the below error in server log


[2018-12-12T16:53:10.777-0500] [glassfish 4.1] [SEVERE] [] [javax.enterprise.resource.webcontainer.jsf.context] [tid: _ThreadID=32 _ThreadName=http-listener-1(4)] [timeMillis: 1544651590777] [levelValue: 1000] [[
  javax.faces.component.UpdateModelException: javax.el.PropertyNotWritableException: /mydata_fragment.xhtml @35,79 value="#{bundle['mydata.result']}": ResourceBundles are immutable
	at javax.faces.component.UIInput.updateModel(UIInput.java:866)
	at javax.faces.component.UIInput.processUpdates(UIInput.java:749)
	at javax.faces.component.UIComponentBase.processUpdates(UIComponentBase.java:1286)
	at org.primefaces.component.api.UITabPanel.processUpdates(UITabPanel.java:1108)
	at org.primefaces.component.tabview.TabView.processUpdates(TabView.java:333)


And it doesn't throw error , if I click "Edit Account -> Account Information or Password" from this link  http://localhost:8080/dataverseuser.xhtml?selectTab=accountInfo

**Comment by JayanthyChengan** [comment]
@scolapasta 
I merged the "alert messages branch" which has jsf:view in all pages, and tested it, but still the language set to "en" and found out that mydata.js calls this below api to get data
http://localhost:8080/api/v1/mydata/retrieve?selected_page=1&dvobject_types=Dataverse&dvobject_types........

And  it hits the condition - FacesContext.getCurrentInstance() == null in the DataverseLocaleBean.java , and set localecode = "en" .  Any suggestions on this?

**Comment by JayanthyChengan** [comment]
@mheppler - Thank you for your investigation and the PR.  I will test the code and let you know soon.

**Comment by pdurbin** [comment]
There are merge conflicts in Bundle_fr.properties but I believe that in the long term this file will be remove in favor of the version at https://github.com/GlobalDataverseCommunityConsortium/dataverse-language-packs

**Comment by JayanthyChengan** [comment]
@mheppler - I tested the code which looks good for me.  Can you please move it to codereview?

Thanks

**Comment by mheppler** [comment]
Good stuff @JayanthyChengan. Thank you. The new `var` approach looks great. I have added a few minor suggestions to improve where the bundle text is located on the page, in relation to previously existing message text, as well as some other minor house cleaning of the loginpage that will tiddy up some long forgotten loose ends.

**Comment by JayanthyChengan** [comment]
@mheppler  - Actually, the new var approach was implemented by you in this pull request https://github.com/scholarsportal/dataverse/pull/29 and added the comment https://github.com/IQSS/dataverse/pull/5351#issuecomment-447125913.   I merged and tested the code, it looks good for me.

Sorry,  I am not clear what other suggestions you are pointing to?

**Comment by mheppler** [comment]
You should be able to see my comments in the [files changed](https://github.com/IQSS/dataverse/pull/5351/files/fbbc1b74f3b4797533f4fae0a7f8f7f92d11d252) view of this PR. It looks like they were "pending" though, as part of a review form, so that was on me for not hitting "submit". You should see them now.

File: src/main/webapp/loginpage.xhtml [hunk 1]
```diff
@@ -157,8 +157,7 @@
 
                                         < href=${URL}?entityID=${EI}&return=${RET}
                                     -->
-
-                                    Your Browser does not support javascript. Please use
+                                    #{bundle['javascriptSupportMessage']}
```
[the remark this answers is not in the document]
Comment (JayanthyChengan) [reply src/main/webapp/loginpage.xhtml]: corrected

File: src/main/java/Bundle_fr.properties [hunk 2]
```diff
@@ -2200,3 +2200,13 @@ passwdVal.passwdReq.letter=lettre
 passwdVal.passwdReq.numeral=chiffre
 passwdVal.passwdReq.special=caract�re sp�cial
 dataretrieverAPI.noMsgResultsFound=D�sol�, aucun r�sultat n'a �t� trouv�.
+
```
[the remark this answers is not in the document]
Comment (JayanthyChengan) [reply src/main/java/Bundle_fr.properties]: corrected

File: src/main/java/Bundle.properties [hunk 3]
```diff
@@ -2294,3 +2294,13 @@ rtabfileparser.ioexception.boolean=Unexpected value for the Boolean variable ({0
 rtabfileparser.ioexception.read=Couldn't read Boolean variable ({0})!
 rtabfileparser.ioexception.parser1=R Tab File Parser: Could not obtain varQnty from the dataset metadata.
 rtabfileparser.ioexception.parser2=R Tab File Parser: varQnty=0 in the dataset metadata!
+
+preloadmessage=(Loading versions...)
+mydata.result=Result
+mydata.results=Results
+mydata.viewnext=View Next
+mydata.more=More
+javascriptSupportMessage=Your Browser does not support javascript. Please use
```
[the remark this answers is not in the document]
Comment (JayanthyChengan) [reply src/main/java/Bundle.properties]: corrected

File: src/main/java/Bundle.properties [hunk 4]
```diff
@@ -2294,3 +2294,13 @@ rtabfileparser.ioexception.boolean=Unexpected value for the Boolean variable ({0
 rtabfileparser.ioexception.read=Couldn't read Boolean variable ({0})!
 rtabfileparser.ioexception.parser1=R Tab File Parser: Could not obtain varQnty from the dataset metadata.
 rtabfileparser.ioexception.parser2=R Tab File Parser: varQnty=0 in the dataset metadata!
+
+preloadmessage=(Loading versions...)
```
[the remark this answers is not in the document]
Comment (JayanthyChengan) [reply src/main/java/Bundle.properties]: corrected

File: src/main/java/Bundle.properties [hunk 5]
```diff
@@ -2294,3 +2294,13 @@ rtabfileparser.ioexception.boolean=Unexpected value for the Boolean variable ({0
...
+mydata.result=Result
```
[the remark this answers is not in the document]
Comment (JayanthyChengan) [reply src/main/java/Bundle.properties]: corrected

**Comment by JayanthyChengan** [comment]
@mheppler - please review the code.  Thanks

**Comment by mheppler** [comment]
@JayanthyChengan thanks for the quick turn around, but I've got two last items...

- In dataverseuser.xhtml `<f:loadBundle basename="Bundle" var="bundle"/>` (line 26) should be removed, it just looks you resorted the initial change
- In Bundle.properties and in Bundle_fr.properties, the `to` and `of` values (line 2209, 2210) are still hanging at the bottom, there is a section of general single words at the top of the bundles, ending at line 71, that they should get moved to

**Comment by JayanthyChengan** [comment]
@mheppler - added new commits.

**Pull request merged**